ESSArch_Core/test_util.py

```diff
@@ -46,3 +57,125 @@ def test_zero_returncode_with_file_created(self, mock_popen, mock_isfile):
 
         cmd = 'unoconv -f %s -eSelectPdfVersion=1 "%s"' % ('pdf', 'test.docx')
         mock_popen.assert_called_once_with(cmd, shell=True, stderr=PIPE, stdout=PIPE)
+
+
+class GetValueFromPathTest(TestCase):
+
+    @staticmethod
+    def get_simple_xml():
+        return '''
+                    <volym andraddat="2014-03-11T14:10:35">
+                        <volnr>1</volnr>
+                        <tid>2000 -- 2005</tid>
+                        <utseende>utseende_text</utseende>
+                        <anmerkningar>anmerkningar_text</anmerkningar>
+                    </volym>
+        '''
+
+    def test_get_value_from_path_when_path_is_none(self):
+        xml = self.get_simple_xml()
+        root_xml = objectify.fromstring(xml)
+        self.assertEqual(get_value_from_path(root_xml, None), None)
+
+    def test_get_value_from_path_when_attribute_is_missing(self):
+        xml = self.get_simple_xml()
+        root_xml = objectify.fromstring(xml)
+        self.assertEqual(get_value_from_path(root_xml, "anmerkningar@onetuhoent"), None)
+
+
+class GetFilesAndDirsTest(TestCase):
+
+    def test_get_files_and_dirs_none_existent_dir(self):
+        self.assertEqual(get_files_and_dirs("this_folder_or_path_should_not_exist"), [])
+
+
+class ParseContentRangeHeaderTest(TestCase):
+
+    def test_parse_content_range_header(self):
+        header = 'bytes 123-456/789'
+        ref_start = 123
+        ref_end = 456
+        ref_total = 789
+
+        (start, end, total) = parse_content_range_header(header)
+        self.assertEqual(start, ref_start)
+        self.assertEqual(end, ref_end)
+        self.assertEqual(total, ref_total)
+
+    def test_parse_content_range_header_with_bad_header(self):
+        header = "bad header"
+        from rest_framework.exceptions import ValidationError
+
+        with self.assertRaises(ValidationError) as context:
+            parse_content_range_header(header)
+
+        self.assertTrue('Invalid Content-Range header' in str(context.exception))
```

Comment:
Use https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex to validate exception message